src/journal/src/Journal/Internal/ByteBufferPtr.hs

```diff
@@ -119,9 +119,9 @@ remaining bb = do
 
 boundCheck :: HasCallStack => ByteBuffer -> Int -> IO ()
```

Comment:
Should we change `boundCheck` so that it returns the real index? (i.e `ix + slice`).?

Reply:
Hmm, not sure, that would make it harder to be able to "turn off" bound checking later on?